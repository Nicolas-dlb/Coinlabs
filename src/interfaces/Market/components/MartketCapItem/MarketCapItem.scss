.market_cap_item {
	display: flex;
	align-items: center;
	justify-content: space-between;
	padding-top: 3.5vh;
	width: 95.8%;
	.item_name {
		width: 180px;
		display: flex;
		align-items: center;
		justify-content: flex-start;

		p {
			margin-left: 5px;
		}
		svg {
			scale: 0.9;
		}
	}
	.item_variation {
		display: flex;
		align-items: center;
		width: 55px;
		margin-left: 5%;
		margin-right: 6%;
		svg {
			width: 15px;
			margin-right: 5px;
		}
		p:first-child {
			margin-right: 5px;
		}
	}
	.item_volume {
		display: flex;
		align-items: center;
		justify-content: flex-end;
		width: 185px;
		p:first-child {
			margin-right: 5px;
		}
		p:last-child {
			font-size: 14px;
		}
	}
}

@media only screen and (min-width: 992px) {
	.market_cap_item {
		padding-top: 2.1vh;
		.item_variation {
			margin-right: 2.5%;
		}
	}
}
