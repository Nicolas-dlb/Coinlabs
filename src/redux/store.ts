import { configureStore } from "@reduxjs/toolkit";
import marketReducer from "./reducers/marketSlice";
import walletsReducer from "./reducers/walletsSlice";
import appReducer from "./reducers/appSlice";
import userReducer from "./reducers/userSlice";

export default configureStore({
  reducer: {
    market: marketReducer,
    wallets: walletsReducer,
    app: appReducer,
    user: userReducer,
  },
});
