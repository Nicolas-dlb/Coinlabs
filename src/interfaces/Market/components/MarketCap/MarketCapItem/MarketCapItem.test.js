import React from "react";
import { render } from "enzyme";
import { Provider } from "react-redux";
import store from "redux/store";
import { act } from "react-test-renderer";
import MarketCapItem from "./MarketCapItem";

describe("MarketCapItem", () => {
  let MarketCapItemComponent;
  beforeEach(async () => {
    await act(async () => {
      MarketCapItemComponent = render(
        <Provider store={store}>
          <MarketCapItem name="bitcoin" />
        </Provider>
      );
    });
  });
  it("should render correctly", () => {
    expect(MarketCapItemComponent).toMatchSnapshot();
  });

  it("should render custom market cap", () => {
    expect(MarketCapItemComponent.find(".item_name").text()).toBe("bitcoin");
  });
});
