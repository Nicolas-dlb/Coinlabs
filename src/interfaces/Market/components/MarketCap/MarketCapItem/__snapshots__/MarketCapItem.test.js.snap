// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketCapItem should render correctly 1`] = `
<div
  class="market_cap_item"
>
  <div
    class="item_name"
  >
    <svg
      height="32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <circle
          cx="16"
          cy="16"
          fill="#2c3a61"
          r="16"
        />
        <g
          fill="#3d97d4"
          fill-rule="nonzero"
        >
          <path
            d="M16.498 4v8.87l7.497 3.35z"
            fill-opacity=".602"
          />
          <path
            d="M16.498 4L9 16.22l7.498-3.35z"
          />
          <path
            d="M16.498 21.968v6.027L24 17.616z"
            fill-opacity=".602"
          />
          <path
            d="M16.498 27.995v-6.028L9 17.616z"
          />
          <path
            d="M16.498 20.573l7.497-4.353-7.497-3.348z"
            fill-opacity=".2"
          />
          <path
            d="M9 16.22l7.498 4.353v-7.701z"
            fill-opacity=".602"
          />
        </g>
      </g>
    </svg>
    <p>
      bitcoin
    </p>
  </div>
  <div
    class="item_variation"
    style="color:#f1c94f"
  >
    <svg
      fill="#f1c94f"
      style="transform:rotate(210deg)"
      viewBox="0 0 492 492"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M464.3 207.4l0.8 0.2H135.9l103.5-103.7c5.1-5.1 7.8-11.9 7.8-19.1 0-7.2-2.8-14-7.8-19.1L223.3 49.5c-5.1-5.1-11.8-7.9-19-7.9 -7.2 0-14 2.8-19 7.8L7.8 226.9C2.8 232 0 238.8 0 246c0 7.2 2.8 14 7.8 19.1l177.4 177.4c5.1 5.1 11.8 7.8 19 7.8 7.2 0 13.9-2.8 19-7.8l16.1-16.1c5.1-5.1 7.8-11.8 7.8-19 0-7.2-2.8-13.6-7.8-18.7L134.7 284.4h330c14.8 0 27.3-12.8 27.3-27.6v-22.8C492 219.2 479.2 207.4 464.3 207.4z"
      />
    </svg>
    <p>
      0
    </p>
    <p>
      %
    </p>
  </div>
  <div
    class="item_volume"
  >
    <p>
      $
    </p>
    <p>
      0 Milliards
    </p>
  </div>
</div>
`;
